python/cudf/cudf/_libxx/join.pyx

```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+
+from cudf._libxx.lib cimport *
+from cudf._libxx.table cimport Table
+from cudf._libxx.join cimport *
+
+from cudf._lib.utils cimport *
+from cudf._lib.utils import *
+
+
+cpdef join(lhs, rhs, left_on, right_on, how, method):
+    """
+      Call libcudf++ join for full outer, inner and left joins.
+      Returns a list of tuples [(column, valid, name), ...]
+    """
+    if how not in ['left', 'inner', 'outer']:
```

Comment:
Semi / anti join in a follow up PR I guess?

Reply:
Yes - I actually already have the code for this, but a few of my tests still lead to segfaults. In some cases libcudf returns the wrong number of columns and so the code fails when we try and construct the output table. I think the reason we're finding all of these now is because the python side of things has a lot broader test coverage.